packages/cadl-ranch-specs/http/routes/main.tsp

```diff
@@ -0,0 +1,476 @@
+    namespace Explode {
+      @scenario
+      @scenarioDoc("""
+        Test path expansion with explode: true when passed a primitive value.
+            Param value: "a"
+            Expected path: /routes/path/explode/primitive/a
+        """)
+      @route("primitive{/param*}")
+      op primitive(param: string): void;
+
+      @scenario
+      @scenarioDoc("""
+        Test path expansion with explode: true when passed an array value.
+            Param value: ["a","b"]
+            Expected path: /routes/path/explode/array/a/b
+        """)
+      @route("array{/param*}")
+      op array(param: string[]): void;
+
+      @scenario
+      @scenarioDoc("""
+        Test path expansion with explode: true when passed a record value.
+            Param value: {a: 1, b: 2}
+            Expected path: /routes/path/explode/record/a=1/b=2
+        """)
+      @route("record{/param*}")
+      op record(param: string[]): void;
+    }
+  }
+
+  @route("label")
+  namespace Label {
+    @route("standard")
+    namespace Standard {
+      @scenario
+      @scenarioDoc("""
+        Test label expansion with explode: false when passed a primitive value.
+            Param value: "a"
+            Expected path: /routes/label/standard/primitive.a
+        """)
+      @route("primitive{.param}")
+      op primitive(param: string): void;
+
+      @scenario
+      @scenarioDoc("""
+        Test label expansion with explode: false when passed an array value.
+            Param value: ["a","b"]
+            Expected path: /routes/label/standard/array.a,b
+        """)
+      @route("array{.param}")
+      op array(param: string[]): void;
+
+      @scenario
+      @scenarioDoc("""
+        Test label expansion with explode: false when passed a record value.
+            Param value: {a: 1, b: 2}
+            Expected path: /routes/label/standard/record.a,1,b,2
+        """)
+      @route("record{.param}")
+      op record(param: string[]): void;
+    }
+
+    @route("explode")
+    namespace Explode {
+      @scenario
+      @scenarioDoc("""
+        Test label expansion with explode: true when passed a primitive value.
+            Param value: "a"
+            Expected path: /routes/label/explode/primitive.a
+        """)
+      @route("primitive{.param*}")
+      op primitive(param: string): void;
+
+      @scenario
+      @scenarioDoc("""
+        Test label expansion with explode: true when passed an array value.
+            Param value: ["a","b"]
+            Expected path: /routes/label/explode/array.a.b
+        """)
+      @route("array{.param*}")
+      op array(param: string[]): void;
+
+      @scenario
+      @scenarioDoc("""
+        Test label expansion with explode: true when passed a record value.
+            Param value: {a: 1, b: 2}
+            Expected path: /routes/label/explode/record.a=1.b=2
+        """)
+      @route("record{.param*}")
+      op record(param: string[]): void;
+    }
+  }
+
+  @route("matrix")
+  namespace Matrix {
```

Comment:
it isn't because there is extra nesting and every scenario is just so slightly different that that main template would still need 7+ params to pass all the different input to output. Could make 2 reusable interface(explode and non explode) but it also feels like it might be more readable with everything explicit anyway here